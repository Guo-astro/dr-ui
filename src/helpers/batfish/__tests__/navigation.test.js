const { buildNavigation } = require('../index.js');
const data = require('./fixtures/data.json');
const dataMulti = require('./fixtures/data-multi.json');
const sections = require('./fixtures/sections.json');

const siteBasePath = '/docs-starter-kit';

describe('buildNavigation', () => {
  it('single structure', () => {
    expect(buildNavigation(siteBasePath, data)).toMatchSnapshot();
  });

  it('multi structure', () => {
    expect(
      buildNavigation(siteBasePath, dataMulti, sections)
    ).toMatchSnapshot();
  });

  it('hideFromNav, removes items', () => {
    expect(
      buildNavigation(siteBasePath, {
        pages: [
          {
            path: '/dr-ui/level-one/',
            frontMatter: {
              title: 'Top page',
              layout: 'accordion',
              navOrder: 1
            }
          },
          {
            path: '/dr-ui/level-one/a/',
            frontMatter: {
              title: 'Page a',
              layout: 'accordion',
              hideFromNav: true
            }
          },
          {
            path: '/dr-ui/level-one/b/',
            frontMatter: {
              title: 'Page b',
              layout: 'accordion'
            }
          }
        ]
      })
    ).toMatchSnapshot();
  });
});
