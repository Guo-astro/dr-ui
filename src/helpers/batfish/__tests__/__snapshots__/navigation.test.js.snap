// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`buildNavigation multi structure 1`] = `
Object {
  "beta/navigation": Object {
    "accordion": Object {
      "/docs-starter-kit/beta/navigation/overview/": Array [
        Object {
          "path": "/docs-starter-kit/beta/navigation/overview/",
          "title": "Overview",
        },
        Object {
          "path": "/docs-starter-kit/beta/navigation/overview/layouts/",
          "title": "Layouts",
        },
        Object {
          "path": "/docs-starter-kit/beta/navigation/overview/navigation/",
          "title": "Navigation",
        },
        Object {
          "path": "/docs-starter-kit/beta/navigation/overview/images/",
          "title": "Images and videos",
        },
        Object {
          "path": "/docs-starter-kit/beta/navigation/overview/constants/",
          "title": "Constants",
        },
        Object {
          "path": "/docs-starter-kit/beta/navigation/overview/frontmatter/",
          "tag": "fundamentals",
          "title": "Frontmatter",
        },
      ],
    },
    "navTabs": Array [
      Object {
        "href": "/docs-starter-kit/beta/navigation/overview/",
        "id": "/docs-starter-kit/beta/navigation/overview/",
        "label": "Overview",
      },
      Object {
        "href": "/docs-starter-kit/beta/navigation/specification/",
        "id": "/docs-starter-kit/beta/navigation/specification/",
        "label": "Specification",
      },
      Object {
        "href": "/docs-starter-kit/beta/navigation/examples/",
        "id": "/docs-starter-kit/beta/navigation/examples/",
        "label": "Examples",
      },
    ],
    "path": "beta/navigation",
    "tag": "beta",
    "title": "Beta Navigation SDK for iOS",
  },
  "hierarchy": Object {
    "/docs-starter-kit/beta/navigation/examples/": Object {
      "parent": "/docs-starter-kit/beta/navigation/examples/",
      "section": Object {
        "path": "beta/navigation",
        "title": "Beta Navigation SDK for iOS",
      },
      "title": "Examples",
    },
    "/docs-starter-kit/beta/navigation/examples/example-1/": Object {
      "parent": "/docs-starter-kit/beta/navigation/examples/",
      "section": Object {
        "path": "beta/navigation",
        "title": "Beta Navigation SDK for iOS",
      },
      "title": "Examples",
    },
    "/docs-starter-kit/beta/navigation/examples/example-2/": Object {
      "parent": "/docs-starter-kit/beta/navigation/examples/",
      "section": Object {
        "path": "beta/navigation",
        "title": "Beta Navigation SDK for iOS",
      },
      "title": "Examples",
    },
    "/docs-starter-kit/beta/navigation/examples/example-3/": Object {
      "parent": "/docs-starter-kit/beta/navigation/examples/",
      "section": Object {
        "path": "beta/navigation",
        "title": "Beta Navigation SDK for iOS",
      },
      "title": "Examples",
    },
    "/docs-starter-kit/beta/navigation/overview/": Object {
      "parent": "/docs-starter-kit/beta/navigation/overview/",
      "section": Object {
        "path": "beta/navigation",
        "title": "Beta Navigation SDK for iOS",
      },
      "title": "Overview",
    },
    "/docs-starter-kit/beta/navigation/overview/constants/": Object {
      "parent": "/docs-starter-kit/beta/navigation/overview/",
      "section": Object {
        "path": "beta/navigation",
        "title": "Beta Navigation SDK for iOS",
      },
      "title": "Overview",
    },
    "/docs-starter-kit/beta/navigation/overview/frontmatter/": Object {
      "parent": "/docs-starter-kit/beta/navigation/overview/",
      "section": Object {
        "path": "beta/navigation",
        "title": "Beta Navigation SDK for iOS",
      },
      "title": "Overview",
    },
    "/docs-starter-kit/beta/navigation/overview/images/": Object {
      "parent": "/docs-starter-kit/beta/navigation/overview/",
      "section": Object {
        "path": "beta/navigation",
        "title": "Beta Navigation SDK for iOS",
      },
      "title": "Overview",
    },
    "/docs-starter-kit/beta/navigation/overview/layouts/": Object {
      "parent": "/docs-starter-kit/beta/navigation/overview/",
      "section": Object {
        "path": "beta/navigation",
        "title": "Beta Navigation SDK for iOS",
      },
      "title": "Overview",
    },
    "/docs-starter-kit/beta/navigation/overview/navigation/": Object {
      "parent": "/docs-starter-kit/beta/navigation/overview/",
      "section": Object {
        "path": "beta/navigation",
        "title": "Beta Navigation SDK for iOS",
      },
      "title": "Overview",
    },
    "/docs-starter-kit/beta/navigation/specification/": Object {
      "parent": "/docs-starter-kit/beta/navigation/specification/",
      "section": Object {
        "path": "beta/navigation",
        "title": "Beta Navigation SDK for iOS",
      },
      "title": "Specification",
    },
    "/docs-starter-kit/maps/examples/": Object {
      "parent": "/docs-starter-kit/maps/examples/",
      "section": Object {
        "path": "maps",
        "title": "Maps SDK for iOS",
      },
      "title": "Examples",
    },
    "/docs-starter-kit/maps/examples/example-1/": Object {
      "parent": "/docs-starter-kit/maps/examples/",
      "section": Object {
        "path": "maps",
        "title": "Maps SDK for iOS",
      },
      "title": "Examples",
    },
    "/docs-starter-kit/maps/examples/example-2/": Object {
      "parent": "/docs-starter-kit/maps/examples/",
      "section": Object {
        "path": "maps",
        "title": "Maps SDK for iOS",
      },
      "title": "Examples",
    },
    "/docs-starter-kit/maps/examples/example-3/": Object {
      "parent": "/docs-starter-kit/maps/examples/",
      "section": Object {
        "path": "maps",
        "title": "Maps SDK for iOS",
      },
      "title": "Examples",
    },
    "/docs-starter-kit/maps/overview/": Object {
      "parent": "/docs-starter-kit/maps/overview/",
      "section": Object {
        "path": "maps",
        "title": "Maps SDK for iOS",
      },
      "title": "Overview",
    },
    "/docs-starter-kit/maps/overview/constants/": Object {
      "parent": "/docs-starter-kit/maps/overview/",
      "section": Object {
        "path": "maps",
        "title": "Maps SDK for iOS",
      },
      "title": "Overview",
    },
    "/docs-starter-kit/maps/overview/frontmatter/": Object {
      "parent": "/docs-starter-kit/maps/overview/",
      "section": Object {
        "path": "maps",
        "title": "Maps SDK for iOS",
      },
      "title": "Overview",
    },
    "/docs-starter-kit/maps/overview/images/": Object {
      "parent": "/docs-starter-kit/maps/overview/",
      "section": Object {
        "path": "maps",
        "title": "Maps SDK for iOS",
      },
      "title": "Overview",
    },
    "/docs-starter-kit/maps/overview/layouts/": Object {
      "parent": "/docs-starter-kit/maps/overview/",
      "section": Object {
        "path": "maps",
        "title": "Maps SDK for iOS",
      },
      "title": "Overview",
    },
    "/docs-starter-kit/maps/overview/navigation/": Object {
      "parent": "/docs-starter-kit/maps/overview/",
      "section": Object {
        "path": "maps",
        "title": "Maps SDK for iOS",
      },
      "title": "Overview",
    },
    "/docs-starter-kit/maps/specification/": Object {
      "parent": "/docs-starter-kit/maps/specification/",
      "section": Object {
        "path": "maps",
        "title": "Maps SDK for iOS",
      },
      "title": "Specification",
    },
    "/docs-starter-kit/navigation/examples/": Object {
      "parent": "/docs-starter-kit/navigation/examples/",
      "section": Object {
        "path": "navigation",
        "title": "Navigation SDK for iOS",
      },
      "title": "Examples",
    },
    "/docs-starter-kit/navigation/examples/example-1/": Object {
      "parent": "/docs-starter-kit/navigation/examples/",
      "section": Object {
        "path": "navigation",
        "title": "Navigation SDK for iOS",
      },
      "title": "Examples",
    },
    "/docs-starter-kit/navigation/examples/example-2/": Object {
      "parent": "/docs-starter-kit/navigation/examples/",
      "section": Object {
        "path": "navigation",
        "title": "Navigation SDK for iOS",
      },
      "title": "Examples",
    },
    "/docs-starter-kit/navigation/examples/example-3/": Object {
      "parent": "/docs-starter-kit/navigation/examples/",
      "section": Object {
        "path": "navigation",
        "title": "Navigation SDK for iOS",
      },
      "title": "Examples",
    },
    "/docs-starter-kit/navigation/overview/": Object {
      "parent": "/docs-starter-kit/navigation/overview/",
      "section": Object {
        "path": "navigation",
        "title": "Navigation SDK for iOS",
      },
      "title": "Overview",
    },
    "/docs-starter-kit/navigation/overview/constants/": Object {
      "parent": "/docs-starter-kit/navigation/overview/",
      "section": Object {
        "path": "navigation",
        "title": "Navigation SDK for iOS",
      },
      "title": "Overview",
    },
    "/docs-starter-kit/navigation/overview/frontmatter/": Object {
      "parent": "/docs-starter-kit/navigation/overview/",
      "section": Object {
        "path": "navigation",
        "title": "Navigation SDK for iOS",
      },
      "title": "Overview",
    },
    "/docs-starter-kit/navigation/overview/images/": Object {
      "parent": "/docs-starter-kit/navigation/overview/",
      "section": Object {
        "path": "navigation",
        "title": "Navigation SDK for iOS",
      },
      "title": "Overview",
    },
    "/docs-starter-kit/navigation/overview/layouts/": Object {
      "parent": "/docs-starter-kit/navigation/overview/",
      "section": Object {
        "path": "navigation",
        "title": "Navigation SDK for iOS",
      },
      "title": "Overview",
    },
    "/docs-starter-kit/navigation/overview/navigation/": Object {
      "parent": "/docs-starter-kit/navigation/overview/",
      "section": Object {
        "path": "navigation",
        "title": "Navigation SDK for iOS",
      },
      "title": "Overview",
    },
    "/docs-starter-kit/navigation/specification/": Object {
      "parent": "/docs-starter-kit/navigation/specification/",
      "section": Object {
        "path": "navigation",
        "title": "Navigation SDK for iOS",
      },
      "title": "Specification",
    },
    "/docs-starter-kit/vision/examples/": Object {
      "parent": "/docs-starter-kit/vision/examples/",
      "section": Object {
        "path": "vision",
        "title": "Vision SDK for iOS",
      },
      "title": "Examples",
    },
    "/docs-starter-kit/vision/examples/example-1/": Object {
      "parent": "/docs-starter-kit/vision/examples/",
      "section": Object {
        "path": "vision",
        "title": "Vision SDK for iOS",
      },
      "title": "Examples",
    },
    "/docs-starter-kit/vision/examples/example-2/": Object {
      "parent": "/docs-starter-kit/vision/examples/",
      "section": Object {
        "path": "vision",
        "title": "Vision SDK for iOS",
      },
      "title": "Examples",
    },
    "/docs-starter-kit/vision/examples/example-3/": Object {
      "parent": "/docs-starter-kit/vision/examples/",
      "section": Object {
        "path": "vision",
        "title": "Vision SDK for iOS",
      },
      "title": "Examples",
    },
    "/docs-starter-kit/vision/overview/": Object {
      "parent": "/docs-starter-kit/vision/overview/",
      "section": Object {
        "path": "vision",
        "title": "Vision SDK for iOS",
      },
      "title": "Overview",
    },
    "/docs-starter-kit/vision/overview/constants/": Object {
      "parent": "/docs-starter-kit/vision/overview/",
      "section": Object {
        "path": "vision",
        "title": "Vision SDK for iOS",
      },
      "title": "Overview",
    },
    "/docs-starter-kit/vision/overview/frontmatter/": Object {
      "parent": "/docs-starter-kit/vision/overview/",
      "section": Object {
        "path": "vision",
        "title": "Vision SDK for iOS",
      },
      "title": "Overview",
    },
    "/docs-starter-kit/vision/overview/images/": Object {
      "parent": "/docs-starter-kit/vision/overview/",
      "section": Object {
        "path": "vision",
        "title": "Vision SDK for iOS",
      },
      "title": "Overview",
    },
    "/docs-starter-kit/vision/overview/layouts/": Object {
      "parent": "/docs-starter-kit/vision/overview/",
      "section": Object {
        "path": "vision",
        "title": "Vision SDK for iOS",
      },
      "title": "Overview",
    },
    "/docs-starter-kit/vision/overview/navigation/": Object {
      "parent": "/docs-starter-kit/vision/overview/",
      "section": Object {
        "path": "vision",
        "title": "Vision SDK for iOS",
      },
      "title": "Overview",
    },
    "/docs-starter-kit/vision/specification/": Object {
      "parent": "/docs-starter-kit/vision/specification/",
      "section": Object {
        "path": "vision",
        "title": "Vision SDK for iOS",
      },
      "title": "Specification",
    },
  },
  "maps": Object {
    "accordion": Object {
      "/docs-starter-kit/maps/overview/": Array [
        Object {
          "path": "/docs-starter-kit/maps/overview/",
          "title": "Overview",
        },
        Object {
          "path": "/docs-starter-kit/maps/overview/layouts/",
          "title": "Layouts",
        },
        Object {
          "path": "/docs-starter-kit/maps/overview/navigation/",
          "title": "Navigation",
        },
        Object {
          "path": "/docs-starter-kit/maps/overview/images/",
          "title": "Images and videos",
        },
        Object {
          "path": "/docs-starter-kit/maps/overview/constants/",
          "title": "Constants",
        },
        Object {
          "path": "/docs-starter-kit/maps/overview/frontmatter/",
          "tag": "fundamentals",
          "title": "Frontmatter",
        },
      ],
    },
    "navTabs": Array [
      Object {
        "href": "/docs-starter-kit/maps/overview/",
        "id": "/docs-starter-kit/maps/overview/",
        "label": "Overview",
      },
      Object {
        "href": "/docs-starter-kit/maps/specification/",
        "id": "/docs-starter-kit/maps/specification/",
        "label": "Specification",
      },
      Object {
        "href": "/docs-starter-kit/maps/examples/",
        "id": "/docs-starter-kit/maps/examples/",
        "label": "Examples",
      },
    ],
    "path": "maps",
    "title": "Maps SDK for iOS",
  },
  "navigation": Object {
    "accordion": Object {
      "/docs-starter-kit/navigation/overview/": Array [
        Object {
          "path": "/docs-starter-kit/navigation/overview/",
          "title": "Overview",
        },
        Object {
          "path": "/docs-starter-kit/navigation/overview/layouts/",
          "title": "Layouts",
        },
        Object {
          "path": "/docs-starter-kit/navigation/overview/navigation/",
          "title": "Navigation",
        },
        Object {
          "path": "/docs-starter-kit/navigation/overview/images/",
          "title": "Images and videos",
        },
        Object {
          "path": "/docs-starter-kit/navigation/overview/constants/",
          "title": "Constants",
        },
        Object {
          "path": "/docs-starter-kit/navigation/overview/frontmatter/",
          "tag": "fundamentals",
          "title": "Frontmatter",
        },
      ],
    },
    "navTabs": Array [
      Object {
        "href": "/docs-starter-kit/navigation/overview/",
        "id": "/docs-starter-kit/navigation/overview/",
        "label": "Overview",
      },
      Object {
        "href": "/docs-starter-kit/navigation/specification/",
        "id": "/docs-starter-kit/navigation/specification/",
        "label": "Specification",
      },
      Object {
        "href": "/docs-starter-kit/navigation/examples/",
        "id": "/docs-starter-kit/navigation/examples/",
        "label": "Examples",
      },
    ],
    "path": "navigation",
    "title": "Navigation SDK for iOS",
  },
  "vision": Object {
    "accordion": Object {
      "/docs-starter-kit/vision/overview/": Array [
        Object {
          "path": "/docs-starter-kit/vision/overview/",
          "title": "Overview",
        },
        Object {
          "path": "/docs-starter-kit/vision/overview/layouts/",
          "title": "Layouts",
        },
        Object {
          "path": "/docs-starter-kit/vision/overview/navigation/",
          "title": "Navigation",
        },
        Object {
          "path": "/docs-starter-kit/vision/overview/images/",
          "title": "Images and videos",
        },
        Object {
          "path": "/docs-starter-kit/vision/overview/constants/",
          "title": "Constants",
        },
        Object {
          "path": "/docs-starter-kit/vision/overview/frontmatter/",
          "tag": "fundamentals",
          "title": "Frontmatter",
        },
      ],
    },
    "navTabs": Array [
      Object {
        "href": "/docs-starter-kit/vision/overview/",
        "id": "/docs-starter-kit/vision/overview/",
        "label": "Overview",
      },
      Object {
        "href": "/docs-starter-kit/vision/specification/",
        "id": "/docs-starter-kit/vision/specification/",
        "label": "Specification",
      },
      Object {
        "href": "/docs-starter-kit/vision/examples/",
        "id": "/docs-starter-kit/vision/examples/",
        "label": "Examples",
      },
    ],
    "path": "vision",
    "title": "Vision SDK for iOS",
  },
}
`;

exports[`buildNavigation single structure 1`] = `
Object {
  "accordion": Object {},
  "hierarchy": Object {
    "/dr-ui/": Object {
      "parent": "/dr-ui/",
      "title": "Overview",
    },
    "/dr-ui/404/": Object {
      "parent": "/dr-ui/",
      "title": "Overview",
    },
    "/dr-ui/batfish-helpers/": Object {
      "parent": "/dr-ui/batfish-helpers/",
      "title": "Batfish helpers",
    },
    "/dr-ui/changelog/": Object {
      "parent": "/dr-ui/",
      "title": "Overview",
    },
    "/dr-ui/examples/": Object {
      "parent": "/dr-ui/examples/",
      "title": "Examples",
    },
    "/dr-ui/examples/example-1/": Object {
      "parent": "/dr-ui/examples/",
      "title": "Examples",
    },
    "/dr-ui/examples/example-2/": Object {
      "parent": "/dr-ui/examples/",
      "title": "Examples",
    },
    "/dr-ui/examples/example-3/": Object {
      "parent": "/dr-ui/examples/",
      "title": "Examples",
    },
  },
  "navTabs": Array [
    Object {
      "href": "/dr-ui/",
      "id": "/dr-ui/",
      "label": "Overview",
    },
    Object {
      "href": "/dr-ui/batfish-helpers/",
      "id": "/dr-ui/batfish-helpers/",
      "label": "Batfish helpers",
    },
    Object {
      "href": "/dr-ui/examples/",
      "id": "/dr-ui/examples/",
      "label": "Examples",
    },
  ],
}
`;
